pkg/endpoint/policy.go

```diff
@@ -442,6 +450,11 @@ func (e *Endpoint) RegenerateIfAlive(regenMetadata *regeneration.ExternalRegener
 		state := e.getState()
 		switch state {
 		case StateRestoring, StateWaitingToRegenerate:
+			// Bump the skipped regeneration level if needed so that the existing/queued
+			// regeneration can regenerate on the required level.
+			if regenMetadata.RegenerationLevel > e.skippedRegenerationLevel {
+				e.skippedRegenerationLevel = regenMetadata.RegenerationLevel
```

Comment:
Maybe we should do a similar `logStatusLocked` here but log the priority bump of the regeneration level? It may be useful to know if this logic triggers or not.

Probably `if/else` with the line immediately below.

Reply:
Done!